src/panels/stations/station-list.tsx

```diff
@@ -94,7 +95,7 @@ const StationEntry = (props: { stnId: string; isSelected: boolean; onAction: (ac
 
     return (
         <ListItem className={isSelected ? classes.selectedItem : ''}>
-            {rmgStyle === RmgStyle.GZMTR && <Typography className={classes.listItemNum}>{num}</Typography>}
+            {rmgStyle === RmgStyle.GZMTR || (rmgStyle === RmgStyle.SHMetro && stn_number) && <Typography className={classes.listItemNum}>{num}</Typography>}
```

Comment:
Use parenthesis correctly! You're hiding gzmtr's station number.

`            {(rmgStyle === RmgStyle.GZMTR || (rmgStyle === RmgStyle.SHMetro && showStationNumber)) && (
                <Typography className={classes.listItemNum}>{num}</Typography>
            )}`